plugin/trino-faker/src/main/java/io/trino/plugin/faker/FakerSplitManager.java

```diff
@@ -13,20 +13,46 @@
  */
 package io.trino.plugin.faker;
 
+import com.google.common.collect.ImmutableList;
+import io.trino.spi.HostAddress;
+import io.trino.spi.Node;
+import io.trino.spi.NodeManager;
 import io.trino.spi.connector.ConnectorSession;
+import io.trino.spi.connector.ConnectorSplit;
 import io.trino.spi.connector.ConnectorSplitManager;
 import io.trino.spi.connector.ConnectorSplitSource;
 import io.trino.spi.connector.ConnectorTableHandle;
 import io.trino.spi.connector.ConnectorTransactionHandle;
 import io.trino.spi.connector.Constraint;
 import io.trino.spi.connector.DynamicFilter;
 import io.trino.spi.connector.FixedSplitSource;
+import jakarta.inject.Inject;
 
 import java.util.List;
 
+import static java.lang.Math.ceilDiv;
+import static java.util.Collections.shuffle;
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.toList;
+
 public class FakerSplitManager
         implements ConnectorSplitManager
 {
+    // this is equal to 250 pages generated in FakerPageSource
+    static final long MAX_ROWS_PER_SPLIT = 250 * 4096;
+
+    private final List<HostAddress> addresses;
+
+    @Inject
+    public FakerSplitManager(NodeManager nodeManager)
+    {
+        List<HostAddress> addresses = requireNonNull(nodeManager, "nodeManager is null").getRequiredWorkerNodes().stream()
+                .map(Node::getHostAndPort)
+                .collect(toList());
+        shuffle(addresses);
+        this.addresses = addresses;
```

Comment:
This should move to `getSplits`, ConnectorSplitManager should not keep stale info about worker nodes

Reply:
Agree, as it would also fix the failure in product tests :-)